backend/resolver/aws/asg.go

```diff
@@ -22,20 +24,32 @@ func (r *res) resolveAutoscalingGroupsForInput(ctx context.Context, input proto.
 func (r *res) autoscalingGroupResults(ctx context.Context, region string, ids []string, limit uint32) (*resolver.Results, error) {
 	ctx, handler := resolver.NewFanoutHandler(ctx)
 
-	regions := r.determineRegionsForOption(region)
-	for _, region := range regions {
-		handler.Add(1)
-		go func(region string) {
-			defer handler.Done()
-			groups, err := r.client.DescribeAutoscalingGroups(ctx, region, ids)
-			select {
-			case handler.Channel() <- resolver.NewFanoutResult(groups, err):
-				return
-			case <-handler.Cancelled():
-				return
-			}
-		}(region)
-	}
+	for _, id := range ids {
+		patternValues, ok, err := meta.ExtractPatternValuesFromString((*ec2v1.AutoscalingGroup)(nil), id)
```

Comment:
looking at this again, this call belongs on :138 in the Search function, as it doesn't apply to the Resolve() path, only the Search() path.

i.e.
```golang
case typeURLAutoscalingGroup:
  patternValues, ok, err := meta.ExtractPatternValuesFromString((*ec2v1.AutoscalingGroup)(nil), id)
  if ok {
    return r.autoscalingGroupResults(ctx, patternValues["region"], []string{patternValues["name"]}, limit)
  }
  return r.autoscalingGroupResults(ctx, resolver.OptionAll, []string{query}, limit)
```

same probably goes for other uses of the extract func